src/Raven.Server/Smuggler/Documents/Processors/IndexProcessor.cs

```diff
@@ -142,6 +144,8 @@ private static IndexDefinition ReadLegacyIndexDefinition(BlittableJsonReaderObje
 
             if (indexDefinition.Maps != null)
             {
+                indexDefinition.Maps = indexDefinition.Maps.Select(ReplaceLegacyProperties).ToHashSet();
```

Comment:
Shouldn't use _hash_ set. The user will want to get them in the same order, and if they have duplicates (rare, but possible) we want to maintain them.

Reply:
Talked with @ayende . HashSet is valid at this point. No changes needed here.